backend/src/modules/email-sending/core/application/command/SendEmailCommandHandler.ts

```diff
@@ -0,0 +1,34 @@
+import { CommandHandler } from '../../../../../shared/core/application/command/CommandHandler';
+import { SendEmail } from './SendEmail';
+import { CommandResult } from '../../../../../shared/core/application/command/CommandResult';
+import NodeMailer from 'nodemailer';
+
+export class SendEmailCommandHandler implements CommandHandler<SendEmail> {
```

Comment:
Wszystkie jakieś połączenia z zewnątrznymi systemami robimy w warstwie infrastructure. Tam implementujemy właśnie takie rzeczy jak np. połączenie z bazą danych. W Application - poprzez zdefiniowanie interfejsu mówisz, że do działania aplikacji potrzebujesz coś co będzie wysyłać maile (tak samo jak z repository - mówisz, że potrzebujesz coś co będzie Ci zapisywać / odczytywać dane, nie mówisz w jaki sposób). 
Czyli ten handler dostałby taki interfejs, tak samo jak inne dostają repository. Wtedy też kod jest łatwiej testowalny. Tworząc implementację takie interfejsu, możesz określić np. że korzysta ona z NodeMailera, ale przyjmuje w konstruktorze takie parametry jak konfiguracja przez co będzie wysyłany mail. 
Wtedy tylko w teście wpiszesz konfigurację odnoszącą się do mailhog i voila! :)

Reply:
https://github.com/nowakprojects/CodersCamp2020.Project.FullStack-Node-React.TableSoccerTournaments/pull/37/commits/b070ecfac2a7a1297782e133ac128af439d52be1
coś tu próbowałem wyrzeźbić, ale jednak dalej tego nie czaję. Zamysł ogarniam jak to ma być zrobione (chyba), ale z wykonaniem jest tragedia... poczekam do 19, bo już za dużo czasu nad tym siedzę bez efektu -.-. Nie ogarniam jak mam niby w konstruktorze podać sobie konfigurację...